### PR TITLE
Add functional test of lnd

### PR DESCRIPTION
**Pull request by AnotherDroog**
`install.sh` now runs a functional test of lnd via vagrant, including create_wallet()

Please test `vagrant up` output

### CODE REVIEW

File: compose/neutrino/docker-compose.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: '3.7'
 services:
   lnd:
-    image: lncm/lnd:v0.7.0
+    image: lncm/lnd:v0.7.1-neutrino-wt
```
[the remark this answers is not in the document]
Comment (AnotherDroog) [reply compose/neutrino/docker-compose.yml]: Your branch is out of date

**Comment by AnotherDroog** [comment]
Please update your branch and clean your environment

```vagrant destroy -f && vagrant up```

**Comment by nolim1t** [comment]
this is the output I get now

```
default: Waiting 5s for wallet to be created...
    default: [lncli] rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp [::1]:10009: connect: connection refused"
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

```

**Comment by nolim1t** [comment]
Full logs

```
   default: Waiting 3s for lnd to start up...
    default: Traceback (most recent call last):
    default:   File "/usr/lib/python3.7/site-packages/urllib3/connection.py", line 159, in _new_conn
    default:     (self._dns_host, self.port), self.timeout, **extra_kw)
    default:   File "/usr/lib/python3.7/site-packages/urllib3/util/connection.py", line 80, in create_connection
    default:     raise err
    default:   File "/usr/lib/python3.7/site-packages/urllib3/util/connection.py", line 70, in create_connection
    default:     sock.connect(sa)
    default: ConnectionRefusedError: [Errno 111] Connection refused
    default: 
    default: During handling of the above exception, another exception occurred:
    default: 
    default: Traceback (most recent call last):
    default:   File "/usr/lib/python3.7/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    default:     chunked=chunked)
    default:   File "/usr/lib/python3.7/site-packages/urllib3/connectionpool.py", line 343, in _make_request
    default:     self._validate_conn(conn)
    default:   File "/usr/lib/python3.7/site-packages/urllib3/connectionpool.py", line 839, in _validate_conn
    default:     conn.connect()
    default:   File "/usr/lib/python3.7/site-packages/urllib3/connection.py", line 301, in connect
    default:     conn = self._new_conn()
    default:   File "/usr/lib/python3.7/site-packages/urllib3/connection.py", line 168, in _new_conn
    default:     self, "Failed to establish a new connection: %s" % e)
    default: urllib3.exceptions.NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x7f1dce13ecc0>: Failed to establish a new connection: [Errno 111] Connection refused
    default: 
    default: During handling of the above exception, another exception occurred:
    default: 
    default: Traceback (most recent call last):
    default:   File "/usr/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    default:     timeout=timeout
    default:   File "/usr/lib/python3.7/site-packages/urllib3/connectionpool.py", line 638, in urlopen
    default:     _stacktrace=sys.exc_info()[2])
    default:   File "/usr/lib/python3.7/site-packages/urllib3/util/retry.py", line 399, in increment
    default:     raise MaxRetryError(_pool, url, error or ResponseError(cause))
    default: urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='127.0.0.1', port=8080): Max retries exceeded with url: /v1/initwallet (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f1dce13ecc0>: Failed to establish a new connection: [Errno 111] Connection refused'))
    default: 
    default: During handling of the above exception, another exception occurred:
    default: 
    default: Traceback (most recent call last):
    default:   File "/usr/bin/noma", line 11, in <module>
    default:     load_entry_point('noma', 'console_scripts', 'noma')()
    default:   File "/media/noma/noma/noma.py", line 294, in main
    default:     lnd(args)
    default:   File "/media/noma/noma/noma.py", line 146, in lnd
    default:     lnd.check_wallet()
    default:   File "/media/noma/noma/lnd.py", line 24, in check_wallet
    default:     create_wallet()
    default:   File "/media/noma/noma/lnd.py", line 325, in create_wallet
    default:     URL_INITWALLET, verify=TLS_CERT_PATH, data=dumps(data)
    default:   File "/usr/lib/python3.7/site-packages/requests/api.py", line 116, in post
    default:     return request('post', url, data=data, json=json, **kwargs)
    default:   File "/usr/lib/python3.7/site-packages/requests/api.py", line 60, in request
    default:     return session.request(method=method, url=url, **kwargs)
    default:   File "/usr/lib/python3.7/site-packages/requests/sessions.py", line 533, in request
    default:     resp = self.send(prep, **send_kwargs)
    default:   File "/usr/lib/python3.7/site-packages/requests/sessions.py", line 646, in send
    default:     r = adapter.send(request, **kwargs)
    default:   File "/usr/lib/python3.7/site-packages/requests/adapters.py", line 516, in send
    default:     raise ConnectionError(e, request=request)
    default: requests.exceptions.ConnectionError: HTTPSConnectionPool(host='127.0.0.1', port=8080): Max retries exceeded with url: /v1/initwallet (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f1dce13ecc0>: Failed to establish a new connection: [Errno 111] Connection refused'))
    default: 2019-08-15 10:20:48.072 [INF] LTND: Version: 0.7.1-beta commit=, build=production, logging=default
    default: 2019-08-15 10:20:48.073 [INF] LTND: Active chain: Bitcoin (network=mainnet)
    default: 2019-08-15 10:20:48.160 [INF] CHDB: Checking for schema update: latest_version=9, db_version=9
    default: 2019-08-15 10:20:48.192 [ERR] LTND: Unable to initialize neutrino backend: unable to create neutrino light client: EOF
    default: 2019-08-15 10:20:48.193 [INF] LTND: Shutdown complete
    default: Waiting 5s for wallet to be created...
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

**Comment by nolim1t** [comment]
I changed some of the timeouts in ```install.sh``` and heres what I found

```
 default: 2019-08-15 10:35:18.116 [ERR] LTND: Unable to initialize neutrino backend: unable to create neutrino light client: EOF
    default: 2019-08-15 10:35:18.117 [INF] LTND: Shutdown complete
    default: Unable to initialize neutrino backend: unable to create neutrino light client: EOF
```

**Comment by AnotherDroog** [comment]
A shorter `sleep()` duration might cause problems on slower systems. A more refined `create_wallet()` that signals on completion may be of use.

Please note, it is important to delete `noma/lnd/data/` and similar files before testing.

This is the expected output of `vagrant up`:

```
    default: noma [node management]
    default:
    default: Usage:
    default:   noma (info|start|stop|restart|logs|check|status)
    default:   noma (temp|swap|ram)
    default:   noma (freq|memory|voltage) [<device>]
    default:   noma turbo
    default:   noma usb-setup
    default:   noma install-box
    default:   noma create-swap
    default:   noma tunnel <port> <host>
    default:   noma (backup|restore|source|diff|devtools)
    default:   noma reinstall [--full]
    default:   noma bitcoind (start|stop|info|fastsync|status|check)
    default:   noma bitcoind get <key>
    default:   noma bitcoind set <key> <value>
    default:   noma bitcoind logs
    default:   noma bitcoind rpcauth <username> [<password>]
    default:   noma lnd (start|stop|info)
    default:   noma lnd logs
    default:   noma lnd connect <address>
    default:   noma lnd (create|unlock|status|check)
    default:   noma lnd lncli [<command>...]
    default:   noma lnd get <key> [<section>] [<path>]
    default:   noma lnd set <key> <value> [<section>] [<path>]
    default:   noma lnd autounlock
    default:   noma lnd autoconnect [<path>]
    default:   noma (-h|--help)
    default:   noma --version
    default:
    default: Options:
    default:   -h --help     Show this screen.
    default:   --version     Show version.
    default: .
    default: .
    default: .
    default: .
    default: .
    default: .
    default:
    default: ----------------------------------------------------------------------
    default: Ran 6 tests in 0.025s
    default:
    default: OK
    default: Pulling lnd (lncm/lnd:0.7.1-neutrino-wt)...
    default: 0.7.1-neutrino-wt: Pulling from lncm/lnd
    default: Digest: sha256:5022e02c9f3fa31599b46fa0af58bfae3317b54c20261ede7fdc6f13a839b11d
    default: Status: Downloaded newer image for lncm/lnd:0.7.1-neutrino-wt
    default: Creating neutrino_lnd_1 ...
Creating neutrino_lnd_1 ... done
    default: Waiting 3s for lnd to start up...
    default: Create wallet is successful
    default: 2019-08-17 04:27:59.102 [INF] LTND: Version: 0.7.1-beta commit=, build=production, logging=default
    default: 2019-08-17 04:27:59.103 [INF] LTND: Active chain: Bitcoin (network=mainnet)
    default: 2019-08-17 04:27:59.181 [INF] CHDB: Checking for schema update: latest_version=9, db_version=9
    default: 2019-08-17 04:27:59.218 [INF] RPCS: Generating TLS certificates...
    default: 2019-08-17 04:27:59.232 [INF] RPCS: Done generating TLS certificates
    default: 2019-08-17 04:28:00.854 [INF] BTCN: Loaded 0 addresses from file '/root/.lnd/data/chain/bitcoin/mainnet/peers.json'
    default: 2019-08-17 04:28:00.854 [DBG] BTCN: Starting block notifications subscription manager
    default: 2019-08-17 04:28:00.854 [DBG] BTCN: Waiting for peer connection...
    default: 2019-08-17 04:28:00.854 [INF] BTCN: Registering block subscription: id=1
    default: 2019-08-17 04:28:00.854 [DBG] BTCN: Delivering backlog of block notifications: id=1, start_height=0, end_height=0
    default: 2019-08-17 04:28:00.854 [INF] BTCN: Broadcaster now active
    default: 2019-08-17 04:28:00.854 [DBG] CMGR: Attempting to connect to 13.91.246.35:8333 (reqid 1)
    default: 2019-08-17 04:28:00.855 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
    default: 2019-08-17 04:28:00.855 [DBG] CMGR: Failed to connect to reqid 2: no valid connect address
    default: 2019-08-17 04:28:00.855 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
    default: 2019-08-17 04:28:00.855 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
    default: 2019-08-17 04:28:00.855 [DBG] CMGR: Failed to connect to reqid 3: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 4: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 5: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 6: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 7: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 8: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 9: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 10: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 11: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 12: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 13: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 18: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 19: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 20: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 21: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 22: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 23: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 24: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 25: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 26: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 27: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 28: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 29: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 30: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Max failed connection attempts reached: [25] -- retrying connection in: 5s
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 31: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Max failed connection attempts reached: [25] -- retrying connection in: 5s
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 14: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Max failed connection attempts reached: [25] -- retrying connection in: 5s
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 16: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Max failed connection attempts reached: [25] -- retrying connection in: 5s
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 17: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Max failed connection attempts reached: [25] -- retrying connection in: 5s
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 15: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Max failed connection attempts reached: [25] -- retrying connection in: 5s
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Failed to connect to reqid 32: no valid connect address
    default: 2019-08-17 04:28:00.856 [DBG] CMGR: Max failed connection attempts reached: [25] -- retrying connection in: 5s
    default: 2019-08-17 04:28:01.158 [DBG] CMGR: Connected to 13.91.246.35:8333 (reqid 1)
    default: 2019-08-17 04:28:01.160 [DBG] BTCN: Sending version (agent /btcwire:0.5.0/neutrino:0.0.4-beta/, pver 70013, block 0) to 13.91.246.35:8333 (outbound)
    default: 2019-08-17 04:28:01.465 [DBG] BTCN: Received version (agent /btcwire:0.5.0/btcd:0.12.0/, pver 70013, block 590472) from 13.91.246.35:8333 (outbound)
    default: 2019-08-17 04:28:01.466 [DBG] BTCN: Negotiated protocol version 70013 for peer 13.91.246.35:8333 (outbound)
    default: 2019-08-17 04:28:01.466 [DBG] BTCN: Added time sample of 2s (total: 1)
    default: 2019-08-17 04:28:01.466 [DBG] BTCN: Connected to 13.91.246.35:8333
    default: 2019-08-17 04:28:01.466 [INF] BTCN: New valid peer 13.91.246.35:8333 (outbound) (/btcwire:0.5.0/btcd:0.12.0/)
    default: 2019-08-17 04:28:01.466 [INF] BTCN: Syncing to block height 590472 from peer 13.91.246.35:8333
    default: 2019-08-17 04:28:01.466 [INF] BTCN: Downloading headers for blocks 1 to 11111 from peer 13.91.246.35:8333
    default: 2019-08-17 04:28:01.467 [DBG] BTCN: Received verack from 13.91.246.35:8333 (outbound)
    default: 2019-08-17 04:28:01.467 [DBG] BTCN: Sending getaddr to 13.91.246.35:8333 (outbound)
    default: 2019-08-17 04:28:01.467 [DBG] BTCN: Sending verack to 13.91.246.35:8333 (outbound)
    default: 2019-08-17 04:28:01.468 [DBG] BTCN: Sending getheaders (locator 000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, stop 0000000069e244f73d78e8fd29ba2fd2ed618bd6fa2ee92559f542fdb26e7c1d) to 13.91.246.35:8333 (outbound)
    default: 2019-08-17 04:28:01.468 [DBG] BTCN: New peer 13.91.246.35:8333 (outbound)
    default: 2019-08-17 04:28:01.468 [DBG] BTCN: Peer connected, starting cfHandler.
    default: 2019-08-17 04:28:01.468 [INF] BTCN: Waiting for more block headers, then will start cfheaders sync from height 0...
    default: 2019-08-17 04:28:01.571 [INF] CMGR: DNS discovery failed on seed x49.dnsseed.bluematt.me: lookup x49.dnsseed.bluematt.me on 10.0.2.3:53: no such host
    default: 2019-08-17 04:28:01.598 [INF] CMGR: 25 addresses found from DNS seed seed.bitnodes.io
    default: 2019-08-17 04:28:01.617 [INF] CMGR: DNS discovery failed on seed x49.seed.bitcoinstats.com: lookup x49.seed.bitcoinstats.com on 10.0.2.3:53: no such host
    default: 2019-08-17 04:28:01.952 [DBG] BTCN: Received addr (1000 addr) from 13.91.246.35:8333 (outbound)
    default: Waiting 5s for wallet to be created...
    default: {
    default: 	"version": "0.7.1-beta commit=",
    default: 	"identity_pubkey": "025617cacb7a9aa93e4561f51bed270d3f169790a68ac3bb038f9b4f0347a808ac",
    default: 	"alias": "LNCM BOX Default",
    default: 	"color": "#000000",
    default: 	"num_pending_channels": 0,
    default: 	"num_active_channels": 0,
    default: 	"num_inactive_channels": 0,
    default: 	"num_peers": 0,
    default: 	"block_height": 13000,
    default: 	"block_hash": "00000000d85a4cf33f325d339b43559df5c8aaa1c2af2c71a1255e95f6038b5a",
    default: 	"best_header_timestamp": 2288912640,
    default: 	"synced_to_chain": false,
    default: 	"testnet": false,
    default: 	"chains": [
    default: 		{
    default: 			"chain": "bitcoin",
    default: 			"network": "mainnet"
    default: 		}
    default: 	],
    default: 	"uris": null
    default: }
```

**Pull request merged**